x-pack/plugins/endpoint/public/applications/endpoint/store/routing/action.ts

```diff
@@ -11,4 +11,9 @@ interface UserNavigatedToPage {
   readonly payload: PageId;
 }
 
-export type RoutingAction = UserNavigatedToPage;
+interface UserNavigatedFromPage {
```

Comment:
@peluja1012  would like your feedback here if you get a chance. I added this Routing action, which is dispatched when the component that used `usePageId` is un-mounted. See changes below to the hook.  In Policy List, I'm using it to reset state when the user leaves the view.

Reply:
I think this is good. Though in our next PR we are adding a "routing" middleware, which keeps track of url changes. In the alerts page, we removed `usePageId` in favor of this "routing" middlware. https://github.com/elastic/kibana/pull/57142